cirq-core/cirq/sim/act_on_args.py

```diff
@@ -113,14 +113,14 @@ def _perform_measurement(self, qubits: Sequence['cirq.Qid']) -> List[int]:
         """Child classes that perform measurements should implement this with
         the implementation."""
 
-    def copy(self: TSelf) -> TSelf:
+    def copy(self: TSelf, reuse_buffer: bool = False) -> TSelf:
         """Creates a copy of the object."""
         args = copy.copy(self)
-        self._on_copy(args)
+        self._on_copy(args, reuse_buffer)
```

Comment:
Make sure to do a `if 'reuse_buffer' in inspect.signature(_on_copy)` check here for backwards compatibility, and output a deprecation warning in the `else` branch.

Reply:
I have added the deprecation warning and a test for it, but I am not sure whether I am doing it in the right way. I can continue improving it based on your feedback.